psi4/src/psi4/libfock/COSK.cc

```diff
@@ -344,16 +344,24 @@ void COSK::build_G_component(std::vector<std::shared_ptr<Matrix>>& D, std::vecto
 
     // per-thread K Matrix buffers (for accumulating thread contributions to K)
     std::vector<std::vector<SharedMatrix>> KT(njk, std::vector<SharedMatrix>(nthreads_));
+    std::vector<std::vector<SharedMatrix>> KTgrad(njk, std::vector<SharedMatrix>(nthreads_));
```

Comment:
should the initialization be conditional on the gradient?

Reply:
If you define a variable within an if-statement its scope will be limited to that if-statement block. So not possible to make this conditional on `do_gradient`.

Reply:
Well you could declare it but only initialize it if `do_gradient`. However, the size is `njk*nthreads_*sizeof(std::vector)` which is probably small enough to ignore, since the other way does involve more complicated code.